Sources/XMTPiOS/Client.swift

```diff
@@ -95,28 +108,64 @@ public final class Client: Sendable {
 			)
 			return try await create(account: account, apiClient: apiClient, options: options)
 		} catch {
-			throw ClientError.creationError(error.localizedDescription)
+			throw ClientError.creationError("\(error)")
+		}
+	}
+
+	static func initV3Client(address: String, options: ClientOptions?, source: LegacyIdentitySource, privateKeyBundleV1: PrivateKeyBundleV1) async throws -> FfiXmtpClient? {
+		let v3Client: FfiXmtpClient?
+
+		if options?.enableAlphaMLS == true && options?.api.env == .local {
+			let dbURL = URL.documentsDirectory.appendingPathComponent("xmtp-\(options?.api.env.rawValue ?? "")-\(address).db3")
+			v3Client = try await LibXMTP.createClient(
+				logger: XMTPLogger(),
+				host: GRPCApiClient.envToUrl(env: options?.api.env ?? .local),
+				isSecure: (options?.api.env ?? .local) != .local,
+				db: dbURL.path,
+				encryptionKey: options?.mlsEncryptionKey,
+				accountAddress: address,
+				legacyIdentitySource: source,
+				legacySignedPrivateKeyProto: try privateKeyBundleV1.toV2().identityKey.serializedData()
+			)
+		} else {
+			v3Client = nil
 		}
+
+		return v3Client
 	}
 
 	static func create(account: SigningKey, apiClient: ApiClient, options: ClientOptions? = nil) async throws -> Client {
-		let privateKeyBundleV1 = try await loadOrCreateKeys(for: account, apiClient: apiClient, options: options)
+		let (privateKeyBundleV1, source) = try await loadOrCreateKeys(for: account, apiClient: apiClient, options: options)
+
+		let v3Client = try await initV3Client(
+			address: account.address,
+			options: options,
+			source: source,
+			privateKeyBundleV1: privateKeyBundleV1
+		)
 
-		let client = try Client(address: account.address, privateKeyBundleV1: privateKeyBundleV1, apiClient: apiClient)
+		if let textToSign = v3Client?.textToSign() {
+			let signature = try await account.sign(message: textToSign).rawData
+			try await v3Client?.registerIdentity(recoverableWalletSignature: signature)
+		}
+
+		let client = try Client(address: account.address, privateKeyBundleV1: privateKeyBundleV1, apiClient: apiClient, v3Client: v3Client)
 		try await client.ensureUserContactPublished()
 
+		for codec in (options?.codecs ?? []) {
+			client.register(codec: codec)
+		}
```

Comment:
Same about the codec code? Was this meant to be committed here?

Reply:
This was actually not working before.